### PR TITLE
[feature] #2228: Add Unauthorized variant to smartcontracts query error

### LINKED ISSUE #2228: Introduce Unauthorized `QueryError` when supplied Account ID is not found
## Description

Currently, if you try to make a query to Iroha with an account that is not found by Iroha, then `QueryError::FindError` is returned. `FindError` seems to be more about something *requested* is not found.

I'd suggest to introduce a new `QueryError` that will share semantics with HTTP 401 "Unauthorized" error.

### PR DESCRIPTION
**Pull request by ales-tsurko**
Closes #2228

### Description of the Change

Added `iroha_core::smartcontracts::query::Error::Unauthorized` to better mirror semantics of unauthorized request (i.e. without an account in this case). Also, in case of querying without an account, **HTTP 401 UNAUTHORIZED** status is returned.

### Issue

Previously, in case of query without an account `FindError::Account` returned. And the corresponding **HTTP 404 NOT_FOUND** status code returned. That appeared to be semantically incorrect.

### Benefits

Now `POST /query` in case of request without an account returns `Error::Unauthorized` and **HTTP 401 UNAUTHORIZED** status code.

### Alternate Designs *[optional]*

At first glance, this can be moved to `Error::Permission` and the `Unauthorized` variant could be added to `DenialReason`, but it doesn't share semantics with **HTTP 401 UNAUTHORIZED** (it's **HTTP 403 FORBIDDEN** instead, [which is different](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403)).

### CODE REVIEW

**Review by appetrosyan — DISMISSED** [review]
After merge add it to #api_changes

File: core/src/wsv.rs
Comment (Erigara) [inline core/src/wsv.rs:139]: commented code

**Review by Erigara — CHANGES_REQUESTED** [review]
Will approve after removing commented code

File: core/src/wsv.rs
Comment (pesterev) [inline core/src/wsv.rs:569]: it looks strange to me that not being able to find the account means the account is not authorized.

File: core/src/smartcontracts/isi/account.rs
Comment (pesterev) [inline core/src/smartcontracts/isi/account.rs:44]:
```suggestion
                    QueryError::Find(find_err) if matches!(*find_err, FindError::Asset(_)) => {
                        assert_can_register(&asset_id.definition_id, wsv, self.object.value())?;
                        wsv.asset_or_insert(asset_id, self.object.value().clone())
                            .expect("Account exists");
                        Ok(())
                    }
```

**Review by pesterev — COMMENTED** [review]
LGTM but left comment that suggests one minor improvement

**Review by Erigara — APPROVED** [review]

**Pull request merged**
